src/iosource/Packet.h

```diff
@@ -148,48 +160,142 @@ class Packet {
      */
     static constexpr const u_char L2_EMPTY_ADDR[L2_ADDR_LEN] = {0};
 
-    // These are passed in through the constructor.
-    std::string tag;              /// Used in serialization
-    double time;                  /// Timestamp reconstituted as float
-    pkt_timeval ts;               /// Capture timestamp
-    const u_char* data = nullptr; /// Packet data.
-    uint32_t len;                 /// Actual length on wire
-    uint32_t cap_len;             /// Captured packet length
-    uint32_t link_type;           /// pcap link_type (DLT_EN10MB, DLT_RAW, etc)
+    struct VlanTag {
+        uint16_t id = 0;
+        uint8_t pcp = 0;
+        bool dei = false;
+
+        auto operator<=>(const VlanTag&) const = default;
+    };
+
+    // Accessor methods for VLAN fields
+    // Suppress deprecation warnings since these methods need to access deprecated fields
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#elif defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable : 4996)
+#endif
+    /**
+     * (Outermost) VLAN ID if present, otherwise empty
+     */
+    std::optional<uint32_t> GetVlan() const { return vlan_present ? std::optional{vlan} : std::nullopt; }
 
     /**
-     * If layer 2 is Ethernet, innermost ethertype field.
+     * (Outermost) VLAN tag if present, otherwise empty
      */
-    uint32_t eth_type;
+    std::optional<VlanTag> GetVlanTag() const {
+        if ( ! vlan_present )
+            return {};
+
+        if ( ! std::in_range<uint16_t>(vlan) )
+            reporter->InternalError("VLAN ID value %u exceeds 16 bits", vlan);
```

Comment:
Would expect the checks more in the Set* than the Get* methods? Any reason for delaying the check to Get*?

Reply:
My thought process was that internally, we already hold these guarantees when using the `Set*` methods because the `VlanTag` struct's types are proper. But, a user can then go in and change `vlan` to be `> UINT16_MAX` (via direct field access). Then, Zeek's core may try to get the tag, and that should be an internal error. 

When we deprecate direct access to `vlan` and `inner_vlan`, then these checks are not necessary since they will always use the `Set*` method, which takes in the correct types.

With that being said, this change could do with a "deprecation guide" since there are a couple moving parts to clean up